CMakeLists.txt

```diff
@@ -22,10 +22,13 @@ set(minZlibVersion 1.2.11)
 include(CheckCXXSourceCompiles) # TODO: migrate to CheckSourceCompiles in CMake >= 3.19
 include(FeatureSummary)
 include(FeatureOptionsSetup)
+
 feature_option(STACKTRACE "Enable stacktraces" ON)
 feature_option(GUI "Build GUI application" ON)
 feature_option(WEBUI "Enables built-in HTTP server for headless use" ON)
 feature_option(VERBOSE_CONFIGURE "Show information about PACKAGES_FOUND and PACKAGES_NOT_FOUND in the configure output (only useful for debugging the CMake build scripts)" OFF)
+feature_option(DEFINE_COMMIT_HASH "Defines git commit hash in version.h, WARNING: may increase incremental build times" OFF)
```

Comment:
IMO the variable starting with "DEFINE" looks weird on the command line: `-DDEFINE_COMMIT_HASH`. The first `-D` is already "define" in cmake terminology.
Maybe consider `SHOW_COMMIT_HASH_IN_VERSION` or something like that.